### ISSUE TITLE
conch server OverflowError while connecting, no matter if ssh client is OpenSSH or conch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7611]
|<img alt="dotz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| dotz reported|
|-|-|
|Trac ID|trac#408|
|Type|enhancement|
|Created|2003-11-20 02:33:15Z|

<details><summary>Searchable metadata</summary>

```
trac-id__408 408
type__enhancement enhancement
reporter__dotz dotz
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1069295595000000 1069295595000000
changetime__1469458059937038 1469458059937038
version__ 
owner__ 
cc__z3p cc__dotz
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="dotz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|dotz commented|
|-|-|

```
#!html
<pre>
Using Twisted 1.1.0 and 
2p3 (#1, Sep 17 2003, 16:48:22)  
[GCC 2.95.3 20010315 (release) (NetBSD nb3)] 
 
doc@mainframe:~> uname -a 
NetBSD mainframe.w.lub.pl 1.6.1 NetBSD 1.6.1 (GENERIC) #0: Tue Apr  8 12:05:52 UTC 2003     
autobuild@...:/autobuild/netbsd-1-6/i386/OBJ/autobuild/netbsd-1-6/src/sys/arch/i386/compile/GENERIC 
i386 
 
Host keys generated by OpenSSH 3.7.1.1 (simply try to connect to lublin.t1.pl:22 to get 
fingerprint) 
 
While connecting, no matter if I use conch client or OpenSSH, I get: 
 
[subtract 5 from each line number, I've added some print to get debugging information] 
 
2003/11/19 21:23 CET [SSHServerTransport,1,127.0.0.1] Traceback (most recent call last): 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/internet/default.py", line 501, in 
doSelect 
            _logrun(selectable, _drdw, selectable, method, dict) 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/python/log.py", line 65, in 
callWithLogger 
            callWithContext({"system": lp}, func, *args, **kw) 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/python/log.py", line 52, in 
callWithContext 
            return context.call({ILogContext: newCtx}, func, *args, **kw) 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/python/context.py", line 32, in 
callWithContext 
            return func(*args,**kw) 
        --- &lt;exception caught here> --- 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/internet/default.py", line 510, in 
_doReadOrWrite 
            why = getattr(selectable, method)() 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/internet/tcp.py", line 250, in doRead 
            return self.protocol.dataReceived(data) 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/conch/ssh/transport.py", line 199, in 
dataReceived 
            f(packet[1:]) 
          File "/usr/pkg/lib/python2p3/site-packages/twisted/conch/ssh/transport.py", line 398, in 
ssh_KEX_DH_GEX_INIT 
            while tries &lt; 10 and y &lt; minimum: 
        exceptions.OverflowError: long int too large to convert to float 
         
2003/11/19 21:23 CET [SSHServerTransport,1,127.0.0.1] connection lost 
 
DEBUGGING INFORMATION: 
 
tries is 0 
y is about 617-digit long integer value. print prints it without "L" at the end 
minimum is float, 2047.0 
 
POSSIBLE FIX: 
If I convert minimum to int, it works - but I am not SSH expert at all, I don't know if it's right or 
wrong. 
 
(then it gives me some other error, that I yet have to investigate and maybe send other PR)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
This is fixed in CVS.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset b7f3cfc3ee270ef33f0f8060c1f3bf1c2ab6395d
```
#!CommitTicketReference repository="" revision="b7f3cfc3ee270ef33f0f8060c1f3bf1c2ab6395d"
Merge pull request #408 from twisted/8668-rodrigc-tpfile-py3

Author: rodrigc
Reviewer: alex
Fixes: #8668

Port twisted.test.test_tpfile to Python 3
```
